### PR TITLE
K8SPS-13: Make orchestrator discover MySQL pods 

### PR DESCRIPTION
**Pull request by egegunes**
Before, we were using peer-list utility to discover new MySQL
pods. Although it was working, it wasn't really reliable and crashed
often.

With these changes, we discover pods by sending requests to Orchestrator
from the operator itself. Unfortunately, the assumption that Orchestrator
will discover every instance if we give it only one host is wrong. So,
we need to send a request for every MySQL pod.

Also some basic status logic that checks ready pods against desired number is in this PR.

### CODE REVIEW

File: pkg/database/orchestrator/client/client.go [hunk 1]
```diff
@@ -62,6 +63,26 @@ func New(host string) *orcClient {
 	}
 }
 
+func (o *orcClient) Discover(host string, port int32) (*instance, error) {
```
Comment (delgod) [inline pkg/database/orchestrator/client/client.go:66]: I feel that we agreed that we will add only pod-0 into the Orchestrator (via config or Startup probe).

File: pkg/cluster/status.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	cr.Status.MySQL = mysqlStatus
+
+	o := orchestrator.New(cr)
+	orcStatus, err := r.appStatus(cr, o)
+	if err != nil {
+		return errors.Wrap(err, "get Orchestrator status")
+	}
+	cr.Status.Orchestrator = orcStatus
+
+	return r.writeStatus(cr)
```
Comment (delgod) [inline pkg/cluster/status.go:36]: I don't feel like we calculate the whole status correctly here.
e.g. is CR ready if all pods are running, but MySQL pods are not added to Orchestrator or when no Primary at this moment, etc.

**Review by delgod — CHANGES_REQUESTED** [review]

File: pkg/database/orchestrator/client/client.go (hunk 1 above)
Comment (egegunes) [reply pkg/database/orchestrator/client/client.go:66]: please see the PR description:

> With these changes, we discover pods by sending requests to Orchestrator
from the operator itself. Unfortunately, the assumption that Orchestrator
will discover every instance if we give it only one host is wrong. So,
we need to send a request for every MySQL pod.

File: pkg/cluster/status.go (hunk 2 above)
Comment (egegunes) [reply pkg/cluster/status.go:36]: Actually, we don't provide CR status yet. We just update status for `orchestrator` and `mysql` if they reached the target size.

**Comment by egegunes** [comment]
We discussed with @hors and decided to keep `peer-list`. I'll remove the part where we discover from operator and just keep status logic.

**Pull request closed without merging**